sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -582,6 +582,109 @@ case class StringLocate(substr: Expression, str: Expression, start: Expression)
     }
   }
 
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    if (start.foldable) {
```

Comment:
I was following https://github.com/apache/spark/pull/7779#discussion_r35945742, if it's not required, I think 7779 is fine.